synchronization/src/main/java/de/cyface/synchronization/SyncAdapter.java

```diff
@@ -124,6 +125,12 @@ public void onPerformSync(@NonNull final Account account, @NonNull final Bundle
         // This allows us to mock the #isConnected() check for unit tests
         mockIsConnectedToReturnTrue = extras.containsKey(MOCK_IS_CONNECTED_TO_RETURN_TRUE);
 
+        // If WifiSurveyor.shutdownSurveillance was called in the meantime, cancel sync:
```

Comment:
We only want to have the hard ValidationException below if the sync was not interrupted by us before (e.g. logout), which is a the only legitimate case where the token may be null.